packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -1155,6 +1251,64 @@ describe('<DataGrid /> - Filter', () => {
       render(<TestCase field="brand" operatorValue="contains" />);
       expect(screen.getByRole('textbox', { name: 'Value' }).value).to.equal('');
     });
+    it('should keep filter operator and value if available', async () => {
+      render(
+        <TestCase
+          field="brand"
+          operatorValue="equals"
+          value="Puma"
+          imperativeFilterModel={false}
+        />,
+      );
+      expect(screen.getByRole('textbox', { name: 'Value' }).value).to.equal('Puma');
+      expect(screen.getByRole('combobox', { name: 'Operators' }).value).to.equal('equals');
+      expect(getColumnValues()).to.deep.equal(['Puma']);
+
+      setColumnValue('slogan');
+
+      expect(getColumnValues()).to.deep.equal([]);
+      expect(screen.getByRole('combobox', { name: 'Operators' }).value).to.equal('equals');
+      expect(screen.getByRole('textbox', { name: 'Value' }).value).to.equal('Puma');
+    });
+    it('should reset value if operator is not available for the new column', async () => {
+      render(
+        <TestCase
+          field="brand"
+          operatorValue="contains"
+          value="Pu"
+          imperativeFilterModel={false}
+        />,
+      );
+      expect(screen.getByRole('textbox', { name: 'Value' }).value).to.equal('Pu');
+      expect(screen.getByRole('combobox', { name: 'Operators' }).value).to.equal('contains');
+      expect(getColumnValues()).to.deep.equal(['Puma']);
+
+      setColumnValue('slogan');
+
+      expect(getColumnValues()).to.deep.equal([]);
+      expect(screen.getByRole('combobox', { name: 'Operators' }).value).to.equal('from');
+      expect(screen.getByTestId('customInput').value).to.equal('');
+    });
+    it('should reset value if the new operator has no input component', async () => {
```

Comment:
`async` can be removed if the test doesn't use promises.

```suggestion
    it('should reset value if the new operator has no input component', () => {
```

x other tests